Source/Core/VideoCommon/FramebufferManagerBase.cpp

```diff
@@ -67,12 +68,15 @@ const XFBSourceBase* const* FramebufferManagerBase::GetRealXFBSource(u32 xfbAddr
 	m_realXFBSource->texWidth = fbWidth;
 	m_realXFBSource->texHeight = fbHeight;
 
-	// OpenGL texture coordinates originate at the lower left, which is why
-	// sourceRc.top = fbHeight and sourceRc.bottom = 0.
 	m_realXFBSource->sourceRc.left = 0;
-	m_realXFBSource->sourceRc.top = fbHeight;
+	m_realXFBSource->sourceRc.top = 0;
 	m_realXFBSource->sourceRc.right = fbWidth;
-	m_realXFBSource->sourceRc.bottom = 0;
+	m_realXFBSource->sourceRc.bottom = fbHeight;
+
+	// OpenGL texture coordinates originate at the lower left, which is why
+	// sourceRc.top = fbHeight and sourceRc.bottom = 0.
+	if (g_ActiveConfig.backend_info.APIType == API_OPENGL)
+		std::swap(m_realXFBSource->sourceRc.top, m_realXFBSource->sourceRc.bottom);
```

Comment:
Yes, and that should not be the case. Just because people have been writing spaghetti code and nobody pushed back doesn't give you a pass on writing new spaghetti code.